From ec84c353617e86dcb842a3880d9adf051c0b7f1c Mon Sep 17 00:00:00 2001
From: Vincent Penquerc'h <vincent.penquerch@collabora.co.uk>
Date: Tue, 16 Aug 2011 13:16:22 +0100
Subject: [PATCH] flacparse: fix off by one in frame size check

Yes, I was tracking another bug and the small test file I generated
to test with improbably just happened to trigger this, with a second
and last frame of 1615 bytes.

https://bugzilla.gnome.org/show_bug.cgi?id=656649
---
 gst/audioparsers/gstflacparse.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gst/audioparsers/gstflacparse.c b/gst/audioparsers/gstflacparse.c
index 0a62765..3fe7118 100644
--- a/gst/audioparsers/gstflacparse.c
+++ b/gst/audioparsers/gstflacparse.c
@@ -577,7 +577,7 @@ gst_flac_parse_frame_is_valid (GstFlacParse * flacparse,
   data = GST_BUFFER_DATA (buffer);
   size = GST_BUFFER_SIZE (buffer);
 
-  if (size <= flacparse->min_framesize)
+  if (size < flacparse->min_framesize)
     goto need_more;
 
   header_ret =
